version: '3.8'
services:

  postgresql:
    image: postgres:15
    container_name: postgresql
    environment:
      POSTGRES_DB: hackathon
      POSTGRES_USER: hackathon
      POSTGRES_PASSWORD: hackathon
    ports:
      - "5432:5432"
    volumes:
      - pg_data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U hackathon" ]
      interval: 10s
      timeout: 5s
      retries: 5

  nginx:
    image: nginx:latest
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - ./nginx/public:/usr/share/nginx/html  # Стартовая страница
      - ./nginx/nginx.conf:/etc/nginx/nginx.conf
      - /var/www/certs:/etc/letsencrypt
    networks:
      - default

  php:
    build: ./php
    ports:
      - "8081:8000"  # Изменили на порт встроенного веб-сервера PHP
    depends_on:
      - postgresql
    networks:
      - default

  python:
    build: ./python
    ports:
      - "8080:8000"
    volumes:
      - /var/www/python_env:/app/src/utils_public/.env
    depends_on:
      - postgresql

  go:
    build: ./go
    ports:
      - "8082:8080"
    depends_on:
      - postgresql

  nodejs:
    build: ./nodejs
    ports:
      - "8083:3000"
    depends_on:
      - postgresql

volumes:
  pg_data:
